pkg/tnf/handlers/bootconfig/bootconfig.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package bootconfig
+
+import (
+	"time"
+
+	"github.com/test-network-function/test-network-function/pkg/tnf"
+	"github.com/test-network-function/test-network-function/pkg/tnf/identifier"
+	"github.com/test-network-function/test-network-function/pkg/tnf/reel"
+)
+
+const (
+	successfulOutputRegex = `(?s).+`
+)
+
+// BootConfig holds information regarding boot config in /boot directory.
+type BootConfig struct {
+	bootConfig string // Output variable that stores the boot config as specified in the /boot directory
+	result     int
+	timeout    time.Duration
+	args       []string
+}
+
+// NewBootConfig creates a BootConfig tnf.Test.
+func NewBootConfig(timeout time.Duration, nodeName string) *BootConfig {
+	return &BootConfig{
+		timeout: timeout,
+		result:  tnf.ERROR,
+		args: []string{
+			"echo \"cat /host/boot/loader/entries/\\`ls -l /host/boot/loader/entries/ | sort -r -k 9 | head -n 1 | cut -d\\\" \\\" -f9 -\\`\" | oc debug -q node/" + nodeName,
```

Comment:
@crwr45 I don't understand the question.
And do you have an alternative method of doing this(without running oc debug twice)?
This script essentially checks the "/host/boot/loader/entries/" folder for the file with the highest id and reads it.

Reply:
@crwr45

Reply:
As I said, this is more than I'm comfortable with in a single command. It's hard to understand, both due to the number of steps and the multiple levels of string escaping. It's also going to be hard to debug especially since, as I said above, the regex and code doesn't distinguish between success or errors.
My concern is primarily the readability and debugability of the code with so much complexity embedded in the `BootConfig` command string. If the way to improve this is to have more than one `oc debug` then please consider it, when they are at odds I think readability is more important than brevity.
Another option is to do more of this parsing in Go code, making it more testable.